### PR TITLE
[Refactoring] Move remaining abstract test classes in mailbox api and…

### PR DESCRIPTION
**Pull request by Arsnael**
… their implementations to JUnit 5

### CODE REVIEW

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java [hunk 1]
```diff
@@ -19,41 +19,36 @@
...
-import org.junit.Before;
-import org.junit.Rule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class CassandraMailboxManagerStressTest extends MailboxManagerStressTest<CassandraMailboxManager> {
-    
-    @Rule public DockerCassandraRule cassandraServer = new DockerCassandraRule().allowRestart();
+@ExtendWith(CassandraRestartExtension.class)
```
Comment (trantienduchn) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java]: Not related, I feel the `CassandraRestartExtension` is always coupled with another extension relates to cassandra. Here is `CassandraClusterExtension`. Any idea to simplify it?

Comment (Arsnael) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMailboxManagerStressTest.java]: Not always, only when you want a restart between tests in a same test class... Maybe can do something similar as in the `@Rule` with `allowRestart` ? Should be an other PR though

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**